### DISCUSSION TITLE
COLOR_MODE_FLAG, IMAGE_MODE_FLAG , GRADIENT_MODE_FLAG

### DISCUSSION THREAD

**Discussion by zehawki** [react-page/react-page#895] [Q&A] [answered]
@macrozone could you please let me know what these are and what they control. From here: https://react-page.github.io/docs/#/builtin_plugins?id=react-pageplugins-image

**Comment by macrozone** [comment]
its an outdated example and i removed it.

The background-plugin takes some of these flags. I honestly don't know them, the plugin was made before i started contributing and I personally never used it in a project. I plan to remove these plugins from this repository in the future into its own repository.

**Reply by zehawki** [reply]
But if you remove all the examples, how am I going to figure out how to use all these (superb) plugins?

**Reply by macrozone** [reply]
the example demo project uses this plugin: https://github.com/react-page/react-page/blob/master/examples/plugins/cellPlugins.ts

hope that helps!

**Reply by zehawki** [reply]
The massive usefulness of this project is precisely because of the plugins - and hence starting from a full fledged page builder as shown at https://react-page.github.io/ is of tremendous value

**Reply by zehawki** [reply]
Ah OK, so it all actually works - I should just take direct from the code?

**Reply by macrozone** [reply]
the point with the background plugin is, that its very specific and complex and adds a certain technical debt to the project, that's why i want to have it in its own repository. Its then also easier to add a proper documentation just for that.

In my projects I usually use custom cell Plugins: https://react-page.github.io/docs/#/custom-cell-plugins

the reason is that requirements and design are often set and its in these cases easier to just create a CellPlugin that does exactly what you want. 

Its in particular useful if you already have a  Component in your application that you want to provide to your editors. E.g. imaging you have a `<LoginForm />` component and you want to allow your editors to add that wherever they want, so you can just wrap it in a custom CellPlugin.


-----

having said that: i agree that as long as it is in this repo, we should document it properly

**Comment by zehawki** [comment, marked as answer]
They control the 3 tabs on the background plugin, enabling/disabling each in turn:
1. IMAGE_MODE_FLAG - allows one to add an image as the background
2. COLOR_MODE_FLAG - allows one to set the background color
3. GRADIENT_MODE_FLAG - allows one to create gradients of multiple colors

![image](https://user-images.githubusercontent.com/8095144/108558623-30372480-7320-11eb-954f-21399e5d1453.png)

**Reply by macrozone** [reply]
I will update the docs accordingly.
